src/java.base/share/classes/sun/security/x509/Extension.java

```diff
@@ -172,10 +174,10 @@ public final void encode(OutputStream out) throws IOException {
     @Override
     public void encode(DerOutputStream out) throws IOException {
 
-        if (extensionId == null)
-            throw new IOException("Null OID to encode for the extension!");
-        if (extensionValue == null)
-            throw new IOException("No value to encode for the extension!");
+        Objects.requireNonNull(extensionId,
+                "Null OID to encode for the extension!");
```

Comment:
I would probably drop the exclamation points. And change "Null OID" to "No OID".